### ISSUE TITLE
ATP consumption greater when the cell is static

### ISSUE THREAD

**Issue by 84634E1A607A** [Revolutionary-Games/Thrive#2295]
**Describe the bug**
ATP consumption is greater when the cell is not moving. Strange.

**Screenshots**
![Static](https://user-images.githubusercontent.com/60734649/120881131-a1fe5a00-c601-11eb-95cd-4451ce014c84.png)
Static

![Moving](https://user-images.githubusercontent.com/60734649/120881134-a6c30e00-c601-11eb-88d5-1785c247ffa2.png)
Moving

**System info (please complete the following information):**
0.5.4.0-alpha

Cell:
![image](https://user-images.githubusercontent.com/60734649/120881156-d40fbc00-c601-11eb-99c2-53915803e8d6.png)

**Comment by hhyyrylainen** [comment]
Are you referring to the ATP number being red? That just means it is the limiting factor in the process speed. The fact that it is a process result means that when it is the limiting factor, there's not enough storage. So in this case the ATP being red says "you don't have enough storage for this process to run fast", which in turn means that you aren't using as much ATP when stationary than when moving. So I think everything is working correctly.

**Comment by hhyyrylainen** [comment]
Or are you referring to the glucose numbers (just noticed those)? I think what's happening is that as the process can't run fullspeed it shows the numbers as-if it could run, which has glucose higher, but the lower one shows the actual glucose consumption. I suppose that is a big or at least unoptimal display.

**Comment by 84634E1A607A** [comment]
Actually what I want to say is the amount of glucose consumed. When the cell is static, it's 0.124; when the cell moves forward, it gets down to 0.063. When I move backward or elsewhere the glucose consumption rate is between them. Does that mean when the cell is moving it costs less glucose?

**Comment by hhyyrylainen** [comment]
Probably not. See my later comment: I think it's just a display difference: when a process can't run (the output is red) the input amounts are computed from the full speed process, but when the process can run the input amounts are the actual amounts consumed (which, if the process can't run at full speed, is less than the theoretical maximum). It's pretty hard to display this in a consistent manner.
The reason I coded it this way is that if a process can't run, you couldn't see which compounds you need (and how much) as all the inputs would be just zeroes indicating nothing is consumed...

**Comment by 84634E1A607A** [comment]
But when the cell is moving forward glucose compound (left bottom corner) accumulates faster. That's pretty strange. (I'll post a video soon.)

**Comment by 84634E1A607A** [comment]
Then my suggestion is that we can use (actual rate) / (theoretical rate) instead.

**Comment by hhyyrylainen** [comment]
Where would that be used? Also 0 / 0.1 is still 0. So this still applies:
> The reason I coded it this way is that if a process can't run, you couldn't see which compounds you need (and how much) as all the inputs would be just zeroes indicating nothing is consumed...

**Comment by 84634E1A607A** [comment]
In the cell processes dialog ![image](https://user-images.githubusercontent.com/60734649/120888884-44800280-c62d-11eb-9dcb-65870d788abc.png) use 0 / 0.166 instead?

**Comment by 84634E1A607A** [comment]
And this one ![image](https://user-images.githubusercontent.com/60734649/120888958-945ec980-c62d-11eb-8416-86f6d60ba6c1.png) Actually when my glucose storage is not full Photosynthesis speed is far higher than that.

**Comment by 84634E1A607A** [comment]
https://user-images.githubusercontent.com/60734649/120889075-1bac3d00-c62e-11eb-9671-fbdbcdae6ec2.mp4

Just look at the speed.

**Comment by hhyyrylainen** [comment]
> use 0 / 0.166 instead?
ah you mean like literally? Instead of doing the math. I suppose that could work, unless the UI becomes too convoluted.

So one potential fix to this issue is to just switch over all the numbers to be "{current} / {maximum}" instead of picking current or maximum based on convoluted logic.

**Comment by 84634E1A607A** [comment]
> So one potential fix to this issue is to just switch over all the numbers to be "{current} / {maximum}" instead of picking current or maximum based on convoluted logic.

I mean that.

**Comment by CoreyHendrey** [comment]
Is this still a problem? I've been combing through the code for a while, and cannot seem to find anywhere that the current value is switched to the theoretical maximum.

If it is, could I get repro steps?

**Comment by hhyyrylainen** [comment]
I'm pretty sure none of the code for displaying the speeds has not been changed since this issue was opened. So this should be still relevant. I would assume that replicating the situation from the screenshots and comparing being stationary and moving should still give different numbers to compare.

**Comment by hhyyrylainen** [comment]
@dligr your recent changes to the process panel may have done something to this issue.

**Comment by dligr** [comment]
> [@dligr](https://github.com/dligr) your recent changes to the process panel may have done something to this issue.

Yes, #6838 made the process system report inactive processes as having zero speed, making inputs and outputs show zeroes. I actually intended to do that in #6830 (which is why it is marked as closing #1982).

If we want the old functionality back, it might be pretty easy to do that, I believe (I haven't checked though).

**Comment by hhyyrylainen** [comment]
I was just looking through the old issues and happened to notice this which I noticed because I had the thought that this was changed recently. So thanks for confirming.

I think that as the number display has been changed we can consider this issue done for now. But if there is still some further tweaking this issue / a new one can be opened.
